src/Pulp/Publish.purs

```diff
@@ -84,6 +86,13 @@ action = Action \args -> do
               <> " report this: https://github.com/bodil/pulp/issues/new")))
           pure
 
+checkBowerProject :: AffN Unit
+checkBowerProject = do
+  bower <- FS.exists "bower.json"
+  if bower then pure unit
+    else throwError
+         (error "No bower.json found, publishing is currently only implemented for Bower projects.")
```

Comment:
I'd rather change the wording a little bit here, as it suggests that we simply haven't done the work in Pulp to enable publishing with psc-package-only projects. It's more that I don't think psc-package has enough market share yet to justify Pulp making it easy for people to publish packages without putting them in the bower registry; for the time being, putting your package on the Bower registry should really be considered a mandatory step to publish a package, I think.

Also, I think I'd rather avoid the phrase "Bower projects" as it seems to suggest that they are separate from "psc-package projects". I mean, I guess they are in a sense as locally you would tend to install your dependencies using one or the other but not both. But as a package author, of course you can make your packages available to people using either package manager.